src/ui/common/src/components/workflows/ReactFlowCanvas.tsx

```diff
@@ -19,23 +18,23 @@ type ReactFlowCanvasProps = {
     event: React.MouseEvent,
     element: ReactFlowNode<ReactFlowNodeData>
   ) => void;
-  nodes: ReactFlowNode<ReactFlowNodeData>[];
-  edges: Edge[];
 };
 
 const ReactFlowCanvas: React.FC<ReactFlowCanvasProps> = ({
   onPaneClicked,
   switchSideSheet,
-  nodes,
-  edges,
 }) => {
   const openSideSheetState = useSelector(
     (state: RootState) => state.openSideSheetReducer
   );
-  const { fitView } = useReactFlow();
+  const dagPositionState = useSelector(
+    (state: RootState) => state.workflowReducer.selectedDagPosition
+  );
+
+  const { fitView, viewportInitialized } = useReactFlow();
   useEffect(() => {
     fitView();
-  }, []);
+  }, [dagPositionState]);
 
   useEffect(() => {
     // NOTE(vikram): There's a timeout here because there seems to be a
```

Comment:
timeout still needed?

Reply:
I think this is still necessary. This has to do with a race condition between the CSS transition for the `div` and when the `fitView` call is made.